pilot/pkg/bootstrap/namespacecontroller.go

```diff
@@ -62,82 +60,65 @@ type NamespaceController struct {
+		func(options *metav1.ListOptions) {
+			options.LabelSelector = fields.SelectorFromSet(configMapLabel).String()
+		})
+	configmapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			c.queue.Push(func() error {
+				return c.configMapChange(newObj)
+			})
+		},
+		DeleteFunc: func(obj interface{}) {
+			cm, ok := obj.(*v1.ConfigMap)
+			if !ok {
+				tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 				if !ok {
-					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
-					if !ok {
-						log.Errorf("error decoding object, invalid type")
-						return
-					}
-					cm, ok = tombstone.Obj.(*v1.ConfigMap)
-					if !ok {
-						log.Errorf("error decoding object tombstone, invalid type")
-						return
-					}
+					log.Errorf("error decoding object, invalid type")
+					return
 				}
-				c.queue.Push(func() error {
-					ns, err := core.Namespaces().Get(cm.Namespace, metav1.GetOptions{})
-					if err != nil {
-						return err
-					}
-					// If the namespace is terminating, we may get into a loop of trying to re-add the configmap back
-					// We should make sure the namespace still exists
-					if ns.Status.Phase != v1.NamespaceTerminating {
-						return c.insertDataForNamespace(cm.Namespace)
-					}
-					return nil
-				})
-			},
-		})
-
-	namespaceLW := listwatch.MultiNamespaceListerWatcher([]string{metav1.NamespaceAll}, func(namespace string) cache.ListerWatcher {
-		return &cache.ListWatch{
-			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-				return core.Namespaces().List(options)
-			},
-			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-				return core.Namespaces().Watch(options)
-			}}
+				cm, ok = tombstone.Obj.(*v1.ConfigMap)
+				if !ok {
+					log.Errorf("error decoding object tombstone, invalid type")
+					return
+				}
+			}
+			c.queue.Push(func() error {
+				ns, err := kubeClient.CoreV1().Namespaces().Get(cm.Namespace, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				// If the namespace is terminating, we may get into a loop of trying to re-add the configmap back
+				// We should make sure the namespace still exists
+				if ns.Status.Phase != v1.NamespaceTerminating {
+					return c.insertDataForNamespace(cm.Namespace)
+				}
+				return nil
+			})
+		},
 	})
-	_, c.namespaceController =
-		cache.NewInformer(namespaceLW, &v1.Namespace{}, 0, cache.ResourceEventHandlerFuncs{
-			UpdateFunc: func(_, newObj interface{}) {
-				c.queue.Push(func() error {
-					return c.namespaceChange(newObj)
-				})
-			},
-			AddFunc: func(obj interface{}) {
-				c.queue.Push(func() error {
-					return c.namespaceChange(obj)
-				})
-			},
-		})
-	return c, nil
+	c.configMapController = configmapInformer
+
+	namespaceInformer := informer.NewNamespaceInformer(kubeClient, 0, cache.Indexers{})
+	namespaceInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
```

Comment:
why remove update func?

Reply:
I understand we donot care about ns update